poc/extensions/src/lib.rs

```diff
@@ -0,0 +1,204 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+use core::marker::PhantomData;
+
+use parity_scale_codec::Decode;
+use poc_executor::{XcqExecutor, XcqExecutorContext};
+#[cfg(not(feature = "std"))]
+use scale_info::prelude::{format, string::String};
+pub type XcqResponse = Vec<u8>;
+pub type XcqError = String;
+pub type XcqResult = Result<XcqResponse, XcqError>;
+
+mod dispatchable;
+pub use dispatchable::{DispatchError, Dispatchable};
+mod extension_id;
+pub use extension_id::{ExtensionId, ExtensionIdTy};
+mod error;
+pub use error::ExtensionError;
+mod macros;
+
+mod extension_core;
+mod extension_fungibles;
+
+mod perm_controller;
+pub use perm_controller::{InvokeSource, PermController};
+
+mod guest;
+pub use guest::{Guest, Input, Method};
+
+// alias trait
+trait Extension: Dispatchable + ExtensionId + Decode {}
+impl<T> Extension for T where T: Dispatchable + ExtensionId + Decode {}
+
+trait ExtensionTuple {
+    fn dispatch(extension_id: ExtensionIdTy, data: &[u8]) -> Result<Vec<u8>, ExtensionError>;
+}
+
+struct Context<E: ExtensionTuple, P: PermController> {
+    invoke_source: InvokeSource,
+    phantom_p: PhantomData<P>,
```

Comment:
```suggestion
    _marker: PhantomData<(P, E)>,
```

so we don't need two of it